parser_library/src/debugging/set_symbol_variable.cpp

```diff
@@ -12,95 +12,105 @@
  *   Broadcom, Inc. - initial API and implementation
  */
 
-#include "set_symbol_variable.h"
+#include "context/variables/set_symbol.h"
+#include "variable.h"
 
-#include <cassert>
-#include <stdexcept>
+namespace hlasm_plugin::parser_library::debugging {
+namespace {
+std::string get_string_array_value(const context::set_symbol_base& set_sym);
+std::string get_string_value(const context::set_symbol_base& set_sym, const std::optional<int>& index);
 
-using namespace hlasm_plugin::parser_library;
-using namespace hlasm_plugin::parser_library::debugging;
-
-set_symbol_variable::set_symbol_variable(const context::set_symbol_base& set_sym, int index)
-    : set_symbol_(set_sym)
-    , index_(index)
-    , name_(std::to_string(index))
+std::string get_string_array_value(const context::set_symbol_base& set_sym)
 {
-    value_ = get_string_value(index_);
-}
+    std::string array_value;
+    array_value.append("(");
 
-set_symbol_variable::set_symbol_variable(const context::set_symbol_base& set_sym)
-    : set_symbol_(set_sym)
-    , index_()
-    , name_("&" + set_symbol_.id.to_string())
-{
-    value_ = get_string_value(index_);
-}
+    auto keys = set_sym.keys();
 
-const std::string& set_symbol_variable::get_name() const { return name_; }
+    if (keys.empty())
+    {
+        array_value.append(")");
+        return array_value;
+    }
 
-const std::string& set_symbol_variable::get_value() const { return value_; }
+    for (const auto& key : keys)
+    {
+        array_value.append(get_string_value(set_sym, key));
+        array_value.append(",");
```

Comment:
Looks like it is in the original code as well but I would assume that `,` wouldn't be appended for the last `key`.

Reply:
It is overwritten by the code after the loop.

Reply:
Right 🤦